docs/block-processing/overall.md

```diff
@@ -0,0 +1,105 @@
+inefficient and inconvenient.
+
+For sake of simplicity, we describe state transition function in two parts:
+* Verification: given GState `S` and a main block `B`, check whether `B` is valid (and can be applied to `S`)
+* Modification: given GState `S` and a main block `B` (successfully verified against `S`), produce `S'`
+
+### Block payload
+
+Block payload consist of:
+* `UpdatePayload`. 
+  It's checked by update system component and is described in the [Update system consensus rules](us.md).
+* `TxPayload`. 
+  It's checked by transaction processing component and is described in the [Transaction processing](txp.md).
+* TODO: others 
+
+## Unknown data handling
+
+Many types in Cardano-SL are designed in an extensible way. They can
+be extended via softfork in future (e. g. new data can be attached or
+semantics can be changed). Examples are:
+* `TxIn` has two constructors `TxInUtxo` and `TxInUnknown`. The former
+  is just a reference to an unspent output, it's the only `TxIn` type
+  we use currently. `TxInUnknown` contains arbitrary bytestring as
+  payload. Later we may introduce another type of transaction inputs
+  (e. g. to redeem fees). Then old software will parse these inputs
+  as `TxInUnknown`, but new software will parse them as new type of
+  input (and process them appropriately).
+* `Attributes` is basically a map from 1-byte integers to arbitrary
+  values. Some values can be parsed into known data types, other
+  values are treated as _unparsed fields_. `Attributes` are part of
+  various data types, e. g. `Tx`. In version 0 `Tx` always has empty
+  attributes. But in version 1 we may add more data to `Tx` (put it
+  into `Attributes`). In this case new software will parse this data
+  and old software will treat it as unparsed fields.
+* Unknown address type. Each valid address in Cardano-SL has a
+  type. It can be one of fixed address types or unknown address type
+  which can't be interpreted by current version of software.
+
+There are few more examples, but these three should demonstrate the
+general idea. If we encounter unknown data (unparsed fields, unknown
+tx input type, unknown address type, etc.) in a block, there are two
+possible behaviours:
+1. Consider such block invalid.
+2. Do as many checks as we can and ignore checks which can't be done
+   because data is unknown. This behaviour depends on which type of
+   data we are processing, each particular case is described in more
+   details below.
+
+The behaviour depends on two protocol versions: version used by this
+software and last adopted version. We verify that data in blocks is
+known if protocol version used by this software is greater than or
+equal to the adopted version. That's because in this case:
+
+1. Authors of this software are aware of the adopted version.
+2. Each issued block must be formed with respect to adopted version.
+
+Comparison is quite tricky here. Table below demonstrates it.
```

Comment:
Comparison of what? Can you put more verbose description here? It's not very clear what is "check" and what is being demonstrated.

Reply:
This paragraph may answer your questions:
```
The behaviour depends on two protocol versions: version used by this
software and last adopted version. We verify that data in blocks is
known if protocol version used by this software is greater than or
equal to the adopted version. That's because in this case:
```